File: src/features/cost-codes/dto/update-cost-code.dto.ts
import { Transform } from 'class-transformer';
import { IsOptional, IsString } from 'class-validator';

export class UpdateCostCodeDto {
  @IsOptional()
  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  category?: string;

  @IsOptional()
  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  code?: string;

  @IsOptional()
  @IsString()
  @Transform(({ value }) => (typeof value === 'string' ? value.trim() : value))
  description?: string;
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/cost-codes/UpdateCostCodeDto.java
package com.mytegroup.api.dto.costcodes;

public record UpdateCostCodeDto(
    String category,
    String code,
    String description
) {
    public UpdateCostCodeDto {
        if (category != null) {
            category = category.trim();
        }
        if (code != null) {
            code = code.trim();
        }
        if (description != null) {
            description = description.trim();
        }
    }
}


